.github/workflows/publish.yml

```diff
@@ -0,0 +1,142 @@
+# https://docs.github.com/en/actions/creating-actions/creating-a-composite-action
+name: Publish binaries
+
+on:
+  pull_request:
+    types: [opened, synchronize, closed]
+    branches:
+      - "main"
+      - "master"
+  workflow_dispatch:
+    inputs:
+      distro:
+        description: Distro
+        required: true
+        type: choice
+        options:
+          - debian
+          - alpine
+      arch:
+        description: Arch
+        required: true
+        type: choice
+        options:
+          - aarch64
+          - amd64
+          - armv6hf
+          - armv7hf
+          - i386
+      node-version:
+        description: Node version
+        required: true
+        type: string
+
+env:
+  DISTRO: ${{ inputs.distro || 'debian' }}
```

Comment:
How about defaulting to alpine, since node provides their own builds for debian which afaict is what we atm use in the contracts (rather than building node for debian again on our own).